src/TextInput/index.js

```diff
@@ -14,44 +14,61 @@ const TextInput = React.forwardRef((props, forwardedRef) => {
   const {
     startIcon,
     endIcon,
+    startContent,
+    endContent,
     showClearButton,
     formatter,
     multiline,
     defaultValue,
     onChange,
+    onInputClear = () => {},
     onBlur,
     testId,
     type = "text",
     ...nativeElementProps
   } = props;
 
+  const inputRef = useRef(null);
   const [inputValue, setInputValue] = useState(
     defaultValue ? defaultValue : ""
   );
 
+  // allows us to use the forwarded ref internally
+  useImperativeHandle(forwardedRef, () => inputRef.current);
```

Comment:
This is my first time using this... it allows us to have access to the forwarded ref internally in this component so we may use it to set focus on the input when it's cleared

https://react.dev/reference/react/useImperativeHandle